clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1279,9 +1283,10 @@ public ClientResponse completed(AbstractResponse response, long timeMs) {
                     false, null, null, response);
         }
 
-        public ClientResponse disconnected(long timeMs, AuthenticationException authenticationException) {
+        public ClientResponse disconnected(long timeMs, AuthenticationException authenticationException, boolean timedOut) {
```

Comment:
and maybe we could introduce `timeouted(long timeMs, AuthenticationException authenticationException)` ? if that makes everything more explicit about the timeout situation.

Reply:
Done.